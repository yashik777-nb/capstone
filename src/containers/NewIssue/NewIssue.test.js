import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";

import NewIssue from "./NewIssue";

configure({ adapter: new Adapter() });

const props = {
  addIssue: () => {},
};

const store = {};
const mockStore = configureStore();

const mockFm = jest.fn();

describe("New Issue Form Container Tests", () => {
  let wrapper;

  beforeEach(() => {
    const render = createMount();
    wrapper = render(
      <Provider store={mockStore(store)}>
        <BrowserRouter>
          <NewIssue {...props} onSave={mockFm} />
        </BrowserRouter>
      </Provider>
    );
  });

  it("Should render New Issue Form", () => {
    expect(wrapper.find("IssueForm").length).toEqual(1);
  });

  test("Should Render New Issue Form Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
