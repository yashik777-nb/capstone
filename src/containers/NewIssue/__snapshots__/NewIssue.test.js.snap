// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Issue Form Container Tests Should Render New Issue Form Snapshot 1`] = `ReactWrapper {}`;
